Getting back on track with exception
